processing/src/main/java/org/apache/druid/segment/join/lookup/LookupJoinMatcher.java

```diff
@@ -277,7 +277,7 @@ public void matchRemainder()
     } else {
       currentIterator = Iterators.filter(
           extractor.iterable().iterator(),
-          entry -> !matchedKeys.contains(entry.getKey())
+          entry -> entry != null && !matchedKeys.contains(entry.getKey())
```

Comment:
It doesn't look like this can actually happen, I think.

In theory iterator.next() can return null, but `MapLookupExtractor#iterable` (the only implementation today that does not throw an exception) returns `map.entrySet()` which should never contain a null element.

Reply:
Yeah. It also doesn't seem reasonable that a "row" can be null. I think it's a false warning because of the `Predicate` interface of Guava. It would be better to suppress warning instead of checking null.

Reply:
good call. Looks like this is on a hot path too, so I've added an inspection suppression. 👍